aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -593,5 +594,21 @@ Tensor hann_window(
   return native::hamming_window(
       window_length, periodic, /*alpha=*/0.5, /*beta=*/0.5, options);
 }
+
+template <typename T>
+Tensor tensor(ArrayRef<T> values, const TensorOptions& options) {
+  auto result = at::empty(values.size(), options);
+  for (size_t i = 0; i < values.size(); ++i) {
+    result[i] = values[i];
```

Comment:
I suspect this is going to be pretty slow for any significant size of `values` since `result[i]` creates a new tensor that is a one-element slice of `result`.

Reply:
You're right. I pondered about this for a bit. Can you think of a better way of doing this, such that it also works for CUDA types and so on (i.e. I guess not `memcpy` into the `.data`). Happy to change these things in a follow up PR